modules/management-console-agent/src/main/java/org/apache/ignite/agent/ManagementConsoleProcessor.java

```diff
@@ -220,10 +224,17 @@ public WebSocketManager webSocketManager() {
      * Start agent on local node if this is coordinator node.
      */
     private void launchAgentListener(DiscoveryEvent evt, DiscoCache discoCache) {
-        if (isLocalNodeCoordinator(ctx.discovery())) {
-            cfg = readFromMetaStorage();
+        guard.lock();
 
-            connect();
+        try {
+            if (isLocalNodeCoordinator(ctx.discovery()) && (evt == null || connectPool == null)) {
```

Comment:
Let's declare selfDescriptive variables like
`isCfgChanged = evt == null`
'isAgentNotStarted = connectPool  == null'

And lets check for this flags first before `isLocalNodeCoordinator(ctx.discovery())`